src/main/java/HttpHandleRequest.java

```diff
@@ -0,0 +1,34 @@
+package com.td.HttpServer;
+
+import java.util.*;
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+
+public abstract class HttpHandleRequest {
+
+  private HashMap<String, String> standardHeaders;
+
+  protected final String CODE_200 = "HTTP/1.1 200 OK";
+  protected final String CODE_404 = "HTTP/1.1 404 Not Found";
+  protected final String CODE_400 = "HTTP/1.1 400 Bad Request";
+
+  public abstract HttpResponse generateResponse(HttpRequest request);
+
+  public HttpHandleRequest() {
+    standardHeaders = new HashMap<String, String>();
+    standardHeaders.put("Server", "Simple Http Server written in Java");
+  }
+
+  protected HashMap<String, String> standardHeaders () {
+    HashMap<String, String> rtnHeaders = standardHeaders;
+    rtnHeaders.put("Date", getDateTime());
+    return rtnHeaders;
+  }
```

Comment:
I find it a little strange to have a private variable `standardHeaders` _and_ a protected method `standardHeaders`. It also seems weird to construct the standard headers in two steps, so to speak (i.e. constructing the hashmap in the class constructor (lines 18-19) and then adding another "static"-ish header in the protected method). Why not something simpler like:

```java
public HttpHandleRequest() {
  standardHeaders = setupStandardHeaders();
}

protected HashMap<String, String> standardHeaders () {
  standardHeaders = new HashMap<String, String>();
  standardHeaders.put("Server", "My server");
  standardHeaders.put("Date", getDateTime());
  return standardHeaderes;
}
```